Sources/Ink/Internal/List.swift

```diff
@@ -16,7 +20,24 @@ internal struct List: Fragment {
 
     private static func read(using reader: inout Reader,
                              indentationLength: Int) throws -> List {
-        var list = List(isOrdered: reader.currentCharacter.isNumber)
+        let isOrdered = reader.currentCharacter.isNumber
+    
+        let listMarker: Character
```

Comment:
I think this could benefit from being separated into a utility function, rather than using deferred `let` declarations and the `defer` keyword itself (both of which make the code a bit harder to understand in  my opinion). Together with the enum approach I suggested above, I think that'd make the logic a lot easier to grasp.

Reply:
Where would you put the utility function? Just as a private static function on the List?

Reply:
I did clean up the code there a little bit in the mean time